extension/storage/filestorage/extension.go

```diff
@@ -34,9 +34,9 @@ func newLocalFileStorage(logger *zap.Logger, config *Config) (extension.Extensio
 		} else {
 			dirs = []string{config.Directory}
 		}
-		perm, err := strconv.ParseInt(config.DirectoryPerm, 8, 32)
+		perm, err := strconv.ParseInt(config.DirectoryPermissions, 8, 32)
```

Comment:
The collector should fail on start if given invalid configuration. Can you move this validation of `directory_permissions` to `Config.Validate` method and return an error if the value is invalid?

Reply:
Nice catch. Thanks, I've moved it to Validate()